test/jdk/java/util/concurrent/tck/JSR166TestCase.java

```diff
@@ -1670,10 +1680,12 @@ Thread newStartedThread(Action action) {
      * the thread (in the hope that it may terminate later) and fails.
      */
     void awaitTermination(Thread thread, long timeoutMillis) {
-        try {
-            thread.join(timeoutMillis);
-        } catch (InterruptedException fail) {
-            threadUnexpectedException(fail);
+        for (;;) { // ignore stray interrupts by test harness
+            try {
+                thread.join(timeoutMillis);
+                break;
+            } catch (InterruptedException ignore) {
```

Comment:
@DougLea Shouldn't this deduct the wait-time for the next join if it gets interrupted? 🤔

Reply:
Ordinarily yes, but here it shouldn't matter: the timeout is just an arbitrary value to catch stuck tests, and sometimes waiting twice as long because of an interrupt is OK.